cegui/src/text/LegacyTextParser.cpp

```diff
@@ -216,6 +220,21 @@ bool LegacyTextParser::parse(const String& inText, std::u32string& outText,
     return true;
 }
 
+Sizef getRespectRatioSize(Sizef reqSize, const Sizef& orgSize) {
+	float orgRatio = orgSize.d_width / orgSize.d_height;
+	float reqRatio = reqSize.d_width / reqSize.d_height;
+
+	if (reqSize.d_width == 0) {
+		reqSize.d_width  = reqSize.d_height * orgRatio;
+	} else if (reqSize.d_height == 0 || orgRatio > reqRatio) {
+		reqSize.d_height = reqSize.d_width  / orgRatio;
+	} else /*if (orgRatio < reqRatio)*/ {
+		reqSize.d_width  = reqSize.d_height * orgRatio;
```

Comment:
1. You have two of three branches equivalent. Please consider refactoring this code.
2. Your idea is not clear. A comment might help a reader to get it.
3. Do you absolutely need to strictly compare floats to zero? I see no division on checked values, so most probably you can get rid of this.

Reply:
Zero compare is used to allow `[image-width='100'][aspect-lock='true'][image='test.png']` without setting height for use auto height based on aspect ratio (and the same with height). Otherwise user code must be `[image-height='999999999'][image-width='100'][aspect-lock='true'][image='test.png']`.

And due to this we need three branches (order of test) - first check for zero (auto unlimited height/width), next compare real ratios.

I add some comments to this code.